tree/treeplayer/inc/ROOT/TVec.hxx

```diff
@@ -0,0 +1,574 @@
+   // modifiers
+   void clear() noexcept { fData.clear(); }
+   iterator erase(const_iterator pos) { return fData.erase(pos); }
+   iterator erase(const_iterator first, const_iterator last) { return fData.erase(first, last); }
+   void push_back(T &&value) { fData.push_back(std::forward<T>(value)); }
+   template <class... Args>
+   reference emplace_back(Args &&... args)
+   {
+      fData.emplace_back(std::forward<Args>(args)...);
+      return fData.back();
+   }
+   template <class... Args>
+   iterator emplace(const_iterator pos, Args &&... args)
+   {
+      return fData.emplace(pos, std::forward<Args...>(args...));
+   }
+   void pop_back() { fData.pop_back(); }
+   void resize(size_type count) { fData.resize(count); }
+   void resize(size_type count, const value_type &value) { fData.resize(count, value); }
+   void swap(TVec<T> &other) { std::swap(fData, other.fData); }
+   // arithmetic operators
+   template <typename V>
+   TVec<T> &operator+=(const V &c)
+   {
+      return ROOT::Internal::VecOps::OperateInPlace(*this, [&c](const T &t) { return t + c; });
+   }
+
+   template <typename V>
+   TVec<T> &operator-=(const V &c)
+   {
+      return ROOT::Internal::VecOps::OperateInPlace(*this, [&c](const T &t) { return t - c; });
+   }
+
+   template <typename V>
+   TVec<T> &operator*=(const V &c)
+   {
+      return ROOT::Internal::VecOps::OperateInPlace(*this, [&c](const T &t) { return t * c; });
+   }
+
+   template <typename V>
+   TVec<T> &operator/=(const V &c)
+   {
+      return ROOT::Internal::VecOps::OperateInPlace(*this, [&c](const T &t) { return t / c; });
+   }
+
+   template <typename V>
+   TVec<T> &operator%=(const V &c)
+   {
+      return ROOT::Internal::VecOps::OperateInPlace(*this, [&c](const T &t) { return t % c; });
+   }
+
+   template <typename V>
+   TVec<T> &operator+=(const TVec<V> &v0)
+   {
+      return ROOT::Internal::VecOps::OperateInPlace(*this, v0, "+", [](const T &t, const V &v) { return t + v; });
+   }
+
+   template <typename V>
+   TVec<T> &operator-=(const TVec<V> &v0)
+   {
+      return ROOT::Internal::VecOps::OperateInPlace(*this, v0, "-", [](const T &t, const V &v) { return t - v; });
+   }
+
+   template <typename V>
+   TVec<T> &operator*=(const TVec<V> &v0)
+   {
+      return ROOT::Internal::VecOps::OperateInPlace(*this, v0, "*", [](const T &t, const V &v) { return t * v; });
+   }
+
+   template <typename V>
+   TVec<T> &operator/=(const TVec<V> &v0)
+   {
+      return ROOT::Internal::VecOps::OperateInPlace(*this, v0, "/", [](const T &t, const V &v) { return t / v; });
+   }
+
+   template <typename V>
+   TVec<T> &operator%=(const TVec<V> &v0)
+   {
+      return ROOT::Internal::VecOps::OperateInPlace(*this, v0, "%", [](const T &t, const V &v) { return t % v; });
+   }
+};
+
+/** @name Math Operators with scalars
+ *  Math operators involving TVec
+*/
+///@{
+template <typename T, typename V>
+auto operator+(const TVec<T> &v, const V &c) -> TVec<decltype(v[0] + c)>
```

Comment:
Could you make them friends of TVec, available only for ADL-lookup?

Reply:
Yes, thanks for the suggestion. This would make thinks much easier for the user.